src/bot/middleware/only-with-text.middleware.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * @description
+ * Skip messages without text
+ * */
+function onlyWithText(ctx, next) {
+  /**
+   * Skip messages before bot became admin
+   * */
+  if ((ctx.msg?.date || 0) * 1000 < +ctx.session.botAdminDate) {
+    return;
+  }
+
+  if (ctx.msg.text || ctx.msg.caption) {
```

Comment:
Let's try to create a const `text` that would be equal to `ctx.msg.text || ctx.msg.caption`. Then use it as `if (text)` and then `ctx.msg.text = text`